### PR TITLE
Cicd playbook bitbucket

### PR DESCRIPTION
**Pull request by RotemAmit**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://jira-hq.paloaltonetworks.local/browse/CIAC-2223

## Description
Adding the Bitbucket integration to the CI/CD playbook 

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [X] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/ContentManagement/Playbooks/playbook-Pull_Request_Creation_-_Generic_README.md [hunk 1]
```diff
@@ -5,21 +5,23 @@ This playbook uses the following sub-playbooks, integrations, and scripts.
 
 ### Sub-playbooks
 * Pull Request Creation - Github
+* Pull Request Creation - Bitbucket
 
 ### Integrations
-This playbook does not use any integrations.
+* Bitbucket
```
Comment (Shellyber) [inline Packs/ContentManagement/Playbooks/playbook-Pull_Request_Creation_-_Generic_README.md:11]: Let's add here GitHub as well

File: Packs/ContentManagement/Playbooks/playbook-Pull_Request_Creation_-_Generic_README.md [hunk 2]
```diff
@@ -29,8 +31,12 @@ This playbook does not use any integrations.
 | ChannelName | The channel to which to send notifications. |  | Optional |
 | PullRequestTemplate | Pull request description template. | ### Pull Request created in Cortex XSOAR<br/>**Created by:** {}<br/>**Pack:** {}<br/>**Branch:** {}<br/>**Link to incident in Cortex XSOAR:** {}<br/>{} <br/>--- | Required |
 | MainBranch | The name of the branch you want the changes pulled into, which must be an existing branch on the current repository. | master | Required |
-| GitIntegration | Which version control integration to use. | GitHub | Required |
+| GitIntegration | Which version control integration to use. | bitbucket | Required |
```
Comment (Shellyber) [inline Packs/ContentManagement/Playbooks/playbook-Pull_Request_Creation_-_Generic_README.md:34]: Let's keep the default GitHub

File: Packs/ContentManagement/ReleaseNotes/1_1_5.md [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+#### Playbooks
+##### New: Pull Request Creation - Bitbucket
+- This playbook creates a pull request using Bitbucket integration.
+##### Pull Request Creation - Github
+- Updated the script SuggestBranchName to be generic.
+##### Pull Request Creation - Generic
+- Added the option to create a pull request in Bitbucket.
+
+#### Scripts
+##### New: CommitFilesBitbucket
```
Comment (Shellyber) [inline Packs/ContentManagement/ReleaseNotes/1_1_5.md:14]: This script should be different than Github? 
Is there different functionality?

File: Packs/ContentManagement/ReleaseNotes/1_1_5.md [hunk 4]
```diff
@@ -0,0 +1,18 @@
+
+#### Incident Fields
+- **CI/CD Pull Request Reviewer**
+
+#### Playbooks
+##### New: Pull Request Creation - Bitbucket
+- This playbook creates a pull request using Bitbucket integration.
+##### Pull Request Creation - Github
+- Updated the script SuggestBranchName to be generic.
```
Comment (Shellyber) [inline Packs/ContentManagement/ReleaseNotes/1_1_5.md:9]:
```suggestion
- Updated the script SuggestBranchName to support both GitHub and BitBucket.
```

File: Packs/ContentManagement/Scripts/SuggestBranchName/SuggestBranchName.py [hunk 5]
```diff
@@ -1,31 +1,29 @@
...
 
 ''' MAIN FUNCTION '''
 
 
 def main():
+
     try:
         pack_name = demisto.getArg('pack')
-        branch_name = find_available_branch(pack_name)
+        command_get_branch = demisto.getArg('command_get_branch')
```
Comment (Shellyber) [inline Packs/ContentManagement/Scripts/SuggestBranchName/SuggestBranchName.py:25]:
```suggestion
        command_get_branch = demisto.getArg('use_command')
```

File: Packs/ContentManagement/Scripts/SuggestBranchName/SuggestBranchName.yml [hunk 6]
```diff
@@ -2,6 +2,14 @@ args:
 - description: The name of the pack.
   name: pack
   required: true
+- auto: PREDEFINED
+  description: The name of the pack.
+  name: command_get_branch
```
Comment (Shellyber) [inline Packs/ContentManagement/Scripts/SuggestBranchName/SuggestBranchName.yml:7]:
```suggestion
  name: use_command
```

File: Packs/ContentManagement/Scripts/SuggestBranchName/SuggestBranchName.yml [hunk 7]
```diff
@@ -2,6 +2,14 @@ args:
 - description: The name of the pack.
   name: pack
   required: true
+- auto: PREDEFINED
+  description: The name of the pack.
```
Comment (Shellyber) [inline Packs/ContentManagement/Scripts/SuggestBranchName/SuggestBranchName.yml:6]: Fix the description for the argument

File: Packs/ContentManagement/Scripts/SuggestBranchName/README.md [hunk 8]
```diff
@@ -22,6 +22,7 @@ Pull Request Creation - Generic
 | **Argument Name** | **Description** |
 | --- | --- |
 | pack | The name of the pack. |
+| command_get_branch | The name of the pack. |
```
Comment (Shellyber) [inline Packs/ContentManagement/Scripts/SuggestBranchName/README.md:25]: Fix this according to YML

**Review by Shellyber — CHANGES_REQUESTED** [review]
Please Run Format on the playbooks, there are a lot of fields that need to be fixed.

File: Packs/ContentManagement/Scripts/CommitFiles/CommitFiles.py [hunk 9]
```diff
@@ -141,39 +239,13 @@ def split_yml_file(content_file):
...
-            os.remove(file_object['name'])
-            self.content_type = file_type.value if file_type else file_type
-            folder = ENTITY_TYPE_TO_DIR.get(self.content_type, '')
-            self.path_to_file = os.path.join('Packs', pack_name, folder)
+def commit_git(git_integration: str, branch_name: str, content_file: ContentFile):
+    if git_integration == 'Gitlab':
+        commit_content_item_gitlab(branch_name, content_file)
+    elif git_integration == 'GitHub':
+        commit_content_item(branch_name, content_file)
+    else:
```
Comment (Shellyber) [inline Packs/ContentManagement/Scripts/CommitFiles/CommitFiles.py:247]: Let's just add a comment that the else is for BitBucket

File: Packs/ContentManagement/Scripts/CommitFiles/README.md [hunk 10]
```diff
@@ -11,7 +11,7 @@ This script gets content files as input from the context, commits the files in t
 ## Used In
 ---
 This script is used in the following playbooks and scripts.
-Pull Request Creation - Generic
+* Pull Request Creation - Github
```
Comment (Shellyber) [inline Packs/ContentManagement/Scripts/CommitFiles/README.md:14]: This is used also in BitBucket, let's update as well

File: Packs/ContentManagement/Scripts/SuggestBranchName/README.md [hunk 11]
```diff
@@ -14,14 +14,17 @@ If the branch with the name "MyPack" exists, the script return "MyPack_1".
...
+* Pull Request Creation - Gitlab
+* Pull Request Creation - Bitbucket
 
 ## Inputs
 ---
 
 | **Argument Name** | **Description** |
 | --- | --- |
 | pack | The name of the pack. |
+| use_command | Which command to use. |
```
Comment (Shellyber) [inline Packs/ContentManagement/Scripts/SuggestBranchName/README.md:27]: Let's add to the description the command we support, for readability

**Review by Shellyber — COMMENTED** [review]
Nice Job!

File: Packs/ContentManagement/Scripts/SuggestBranchName/SuggestBranchName.yml [hunk 12]
```diff
@@ -2,6 +2,14 @@ args:
 - description: The name of the pack.
   name: pack
   required: true
+- auto: PREDEFINED
+  description: Which command to use.
+  name: use_command
+  required: true
+  predefined:
+  - gitlab-branch-list
+  - GitHub-get-branch
```
[the remark this answers is not in the document]
Comment (RotemAmit) [reply Packs/ContentManagement/Scripts/SuggestBranchName/SuggestBranchName.yml:11]: yes

File: Packs/ContentManagement/ReleaseNotes/1_1_5.md (hunk 4 above)
Comment (Shellyber) [inline Packs/ContentManagement/ReleaseNotes/1_1_5.md:9]: remove this.

**Review by Shellyber — APPROVED** [review]
Nice job.
Just fix my last comment

**Comment by RotemAmit** [comment]
The test coverage right now is 34.11%, but @maimorag will add more tests to the Gitlub part on the script CommitFiles.py.
 The docker image tag is not the latest numeric tag because it is not working with the latest tag, I opened an issue about that https://jira-hq.paloaltonetworks.local/browse/CIAC-4633 .

**Review by Shellyber — APPROVED** [review]

**Pull request merged**